components/patina_smbios/src/service.rs

```diff
@@ -0,0 +1,389 @@
+/// Platform code typically adds records using `add_record<T>()` and then publishes
+/// the table for the OS to query directly.
+///
+/// # Type Filtering
+///
+/// The iterator can optionally filter by record type. If `None` is provided,
+/// all records are returned. If `Some(type)` is provided, only records of
+/// that type are returned.
+pub struct SmbiosRecordsIter<'a> {
+    records: Ref<'a, Vec<crate::manager::SmbiosRecord>>,
+    position: usize,
+    filter_type: Option<SmbiosType>,
+}
+
+impl<'a> SmbiosRecordsIter<'a> {
+    /// Create a new iterator over SMBIOS records
+    pub(crate) fn new(records: Ref<'a, Vec<crate::manager::SmbiosRecord>>, filter_type: Option<SmbiosType>) -> Self {
+        Self { records, position: 0, filter_type }
+    }
+}
+
+impl<'a> Iterator for SmbiosRecordsIter<'a> {
+    type Item = (SmbiosTableHeader, Option<Handle>);
+
+    fn next(&mut self) -> Option<Self::Item> {
+        while self.position < self.records.len() {
+            let record = &self.records[self.position];
+            self.position += 1;
+
+            // Apply type filter if specified
+            if let Some(filter) = self.filter_type
+                && record.header.record_type != filter
+            {
+                continue;
+            }
+
+            return Some((record.header.clone(), record.producer_handle));
+        }
+
+        None
+    }
+}
+
+/// SMBIOS service providing all SMBIOS operations
+///
+/// This service provides a complete interface for SMBIOS table management:
+/// - **Type-safe operations**: `add_record<T>()` for structured record types (recommended)
+/// - **Record management**: `update_string()`, `remove()`
+/// - **Table management**: `version()`, `publish_table()`
+///
+/// All operations are protected by a TplMutex at TPL_NOTIFY level for thread safety.
+///
+/// # Example
+///
+/// ```ignore
+/// fn entry_point(
+///     smbios: Service<Smbios>,
+/// ) -> Result<()> {
+///     // Add structured records
+///     smbios.add_record(None, &bios_info)?;
+///     
+///     // Publish to configuration table
+///     smbios.publish_table()?;
+///     Ok(())
+/// }
+/// ```
+#[derive(patina::component::service::IntoService, Clone, Copy)]
+#[service(Smbios)]
+pub struct Smbios {
+    pub(crate) manager: &'static patina::tpl_mutex::TplMutex<
+        'static,
+        crate::manager::SmbiosManager,
+        patina::boot_services::StandardBootServices,
+    >,
+    pub(crate) boot_services: &'static patina::boot_services::StandardBootServices,
+    pub(crate) major_version: u8,
+    pub(crate) minor_version: u8,
+}
+
+// Methods below are tested via integration (Q35 platform component)
+// They require StandardBootServices and TplMutex which aren't mockable in unit tests
+impl Smbios {
+    /// Gets the SMBIOS version information.
+    ///
+    /// # Returns
+    ///
+    /// A tuple of (major_version, minor_version).
+    #[cfg_attr(coverage_nightly, coverage(off))]
```

Comment:
very small nit, and you can actually ignore this if you want, but you may be able to move the `#[cfg_attr(coverage_nightly, coverage(off))]` to the `impl Smbios` to ignore coverage for all implementations.